rust/ql/test/library-tests/type-inference/type-inference.ql

```diff
@@ -1,18 +1,60 @@
 import rust
+import utils.test.InlineExpectationsTest
 import codeql.rust.internal.TypeInference as TypeInference
 import TypeInference
-import utils.test.InlineExpectationsTest
 
 query predicate inferType(AstNode n, TypePath path, Type t) {
   t = TypeInference::inferType(n, path)
 }
 
-query predicate resolveMethodCallExpr(MethodCallExpr mce, Function f) {
-  f = resolveMethodCallExpr(mce)
+module ResolveTest implements TestSig {
+  string getARelevantTag() { result = ["method", "field"] }
+
+  private predicate functionHasValue(Function f, string value) {
+    f.getPrecedingComment().getCommentText() = value
+    or
+    not exists(f.getPrecedingComment()) and
+    value = f.getName().getText()
+  }
+
+  predicate hasActualResult(Location location, string element, string tag, string value) {
+    exists(AstNode source, AstNode target |
+      location = source.getLocation() and
+      element = source.toString()
+    |
+      target = resolveMethodCallExpr(source) and
+      functionHasValue(target, value) and
+      tag = "method"
+      or
+      target = resolveStructFieldExpr(source) and
+      any(Struct s | s.getStructField(_) = target).getName().getText() = value and
+      tag = "field"
+      or
+      target = resolveTupleFieldExpr(source) and
+      any(Struct s | s.getTupleField(_) = target).getName().getText() = value and
+      tag = "field"
```

Comment:
I think the name of this tag is a bit confusing.  It says `"field"`, but it seems to be identifying the struct / tuple that `source` is a field *of*.  Should we call it `"fieldof"`???

Reply:
Good idea. I've renamed it to `fieldof`. That definitely reads nicer 👍